internal/core/metadata/operators/device_profile/value_descriptor_test.go

```diff
@@ -0,0 +1,149 @@
+package device_profile
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+
+	"github.com/edgexfoundry/edgex-go/internal/core/metadata/operators/device_profile/mocks"
+)
+
+var TestDeviceResource1 = contract.DeviceResource{
+	Name:        "TestDeviceResource1Name",
+	Description: "TestDeviceResource1Description",
+	Properties: contract.ProfileProperty{
+		Value: contract.PropertyValue{
+			Maximum:       "TestValueMax",
+			Minimum:       "TestValueMin",
+			DefaultValue:  "TestValueDefault",
+			Type:          "TestValueType",
+			FloatEncoding: "TestValueFloatEncoding",
+			MediaType:     "TestValueMediaType",
+		},
+		Units: contract.Units{
+			Type:         "TestUnitType",
+			DefaultValue: "TestDefaultUnitValue",
+			ReadWrite:    "TestReadWrite",
+		},
+	},
+}
+
+var TestDeviceResource2 = contract.DeviceResource{
+	Name:        "TestDeviceResource2Name",
+	Description: "TestDeviceResource2Description",
+	Properties: contract.ProfileProperty{
+		Value: contract.PropertyValue{
+			Maximum:       "TestValueMax",
+			Minimum:       "TestValueMin",
+			DefaultValue:  "TestValueDefault",
+			Type:          "TestValueType",
+			FloatEncoding: "TestValueFloatEncoding",
+			MediaType:     "TestValueMediaType",
+		},
+		Units: contract.Units{
+			Type:         "TestUnitType",
+			DefaultValue: "TestDefaultUnitValue",
+			ReadWrite:    "TestReadWrite",
+		},
+	},
+}
+
+var TestDeviceResourceError = contract.DeviceResource{
+	Name:        "TestDeviceResourceError",
+	Description: "TestDeviceResource1Description",
+	Properties: contract.ProfileProperty{
+		Value: contract.PropertyValue{
+			Maximum:       "TestValueMax",
+			Minimum:       "TestValueMin",
+			DefaultValue:  "TestValueDefault",
+			Type:          "TestValueType",
+			FloatEncoding: "TestValueFloatEncoding",
+			MediaType:     "TestValueMediaType",
+		},
+		Units: contract.Units{
+			Type:         "TestUnitType",
+			DefaultValue: "TestDefaultUnitValue",
+			ReadWrite:    "TestReadWrite",
+		},
+	},
+}
+
+var TestValueDescriptor1 = createValueDescriptorFromDeviceResource(TestDeviceResource1)
+var TestValueDescriptor2 = createValueDescriptorFromDeviceResource(TestDeviceResource2)
+var TestValueDescriptorError = createValueDescriptorFromDeviceResource(TestDeviceResourceError)
+var TestDeviceResources = []contract.DeviceResource{
+	TestDeviceResource1,
+	TestDeviceResource2,
+}
+
+var TestContext = context.Background()
+
+func Test(t *testing.T) {
```

Comment:
Please make the function name more descriptive, such as `TestAddValueDescriptors`